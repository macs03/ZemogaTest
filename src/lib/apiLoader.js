export default class ApiLoader {
  static boot(App) {
    const baseUrl = 'https://jsonplaceholder.typicode.com/';

    App.api = {
      get: (uri) => {
        return fetch(`${baseUrl}${uri}`, {
          method: 'GET',
        });
      },
      post: (uri) => {
        return fetch(`${baseUrl}${uri}`, {
          method: 'POST',
        });
      },
      delete: (uri) => {
        return fetch(`${baseUrl}${uri}`, {
          method: 'DELETE',
        });
      },
    };
  }
}
