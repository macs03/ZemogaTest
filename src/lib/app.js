import AssetsLoader from './assetsLoader';
import ApiLoader from './apiLoader';

const globalLoaders = [AssetsLoader, ApiLoader];

export default class App {
  static api = undefined;

  static boot() {
    return Promise.all(globalLoaders.map((s) => s.boot(App)))
      .then(() => {
        console.info('App is ready!!!');
      })
      .catch((err) => console.error('Error while booting the app', err));
  }
}
