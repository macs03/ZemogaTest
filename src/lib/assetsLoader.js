import {Assets} from 'react-native-ui-lib';

export default class AssetsLoader {
  static boot() {
    Assets.loadAssetsGroup('icons', {
      refresh: require('../assets/icons/refresh.png'),
      favorite: require('../assets/icons/star.png'),
      trash: require('../assets/icons/delete.png'),
      chevron: require('../assets/icons/chevronRight.png'),
    });
  }
}
