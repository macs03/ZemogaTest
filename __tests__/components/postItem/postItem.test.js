/**
 * @format
 */

import 'react-native';
import React from 'react';
import {Assets} from 'react-native-ui-lib';
import PostItem from '../../../src/components/postItem/postItem';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {on, Interactions, findByTestId} from '../../../src/lib/user-simulator';

const item = {
  userId: 1,
  id: 1,
  title:
    'sunt aut facere repellat provident occaecati excepturi optio reprehenderit',
  body:
    'quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto',
  rightItems: {
    text: 'Delete',
    icon: Assets.icons.trash,
    background: '#fff',
    onPress: () => {},
  },
};

it('renders correctly', () => {
  const tree = renderer.create(
    <PostItem
      item={item}
      index={0}
      addRef={() => {}}
      onSwipeableWillOpen={() => {}}
      navigationTo={() => {}}
    />,
  );
  expect(tree).toMatchSnapshot();
});

it('calls navigation after tap an Item', () => {
  const spy = jest.fn();
  const tree = renderer.create(
    <PostItem
      item={item}
      index={0}
      addRef={() => {}}
      onSwipeableWillOpen={() => {}}
      navigationTo={spy}
    />,
  );

  const button = findByTestId(tree, 'navigateToPost');

  on(button).simulate(Interactions.PRESS);

  expect(spy).toHaveBeenCalled();
});
