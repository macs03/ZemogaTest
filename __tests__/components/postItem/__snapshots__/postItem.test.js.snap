// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Drawer
  index={0}
  onSwipeableWillOpen={[Function]}
>
  <ListItem
    height={75.8}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 0.5,
        "borderColor": "#D7D7D7",
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "width": "100%",
      }
    }
    testID="navigateToPost"
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#0E75EE",
            "borderRadius": 4,
            "height": 8,
            "marginHorizontal": 2,
            "width": 8,
          }
        }
      />
    </View>
    <View>
      <Image
        accessibilityRole="image"
        accessible={false}
        assetGroup="icons"
        source={
          Object {
            "testUri": "../../../src/assets/icons/star.png",
          }
        }
        style={
          Array [
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {},
            Object {
              "tintColor": "#FEE21C",
            },
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 17,
          "width": "90%",
        }
      }
    >
      <View
        style={
          Object {
            "marginBottom": 3,
          }
        }
      >
        <Text
          color="#20303C"
          numberOfLines={1}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "textAlign": "left",
              },
              Object {
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              Object {
                "color": "#20303C",
              },
              undefined,
              undefined,
              Object {},
              undefined,
              undefined,
              Object {
                "flex": 1,
                "marginRight": 10,
              },
            ]
          }
          text70={true}
        >
          sunt aut facere repellat provident occaecati excepturi optio reprehenderit
        </Text>
      </View>
    </View>
    <View>
      <Image
        accessibilityRole="image"
        accessible={false}
        assetGroup="icons"
        source={
          Object {
            "testUri": "../../../src/assets/icons/chevronRight.png",
          }
        }
        style={
          Array [
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {},
            Object {
              "tintColor": "#D7D7D7",
            },
          ]
        }
      />
    </View>
  </ListItem>
</Drawer>
`;
